flowzone.yml

```diff
@@ -1529,87 +1535,39 @@ jobs:
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.7"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.7 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.8
+      - name: Set up Python environment
         if: steps.python_poetry.outputs.enabled == 'true'
         uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
         with:
-          python-version: "3.8"
+          python-version: ${{ env.SUPPORTED_VERSIONS }}
 
-      - name: Check compatibility
+      - name: Validate project Python requirements
         if: steps.python_poetry.outputs.enabled == 'true'
         run: |
-          error_check=`(poetry env use 3.8 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.8"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.8 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.9
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.9"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.9 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.9"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.9 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.10
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.10"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.10 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.10"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.10 does not meet project requirements."
-          fi
+          versions=()
+          while IFS= read -r version; do
+            echo "Setting up Python $version"
+            error_check=`(poetry env use $version 2>&1 || true)`
+            if ! grep -q "Please choose a compatible version" <<< $error_check; then
+              versions+=("$version")
+            else
+              echo "Python $version does not meet project requirements."
+            fi
+          done <<< "$(echo -n "${SUPPORTED_VERSIONS}")"
+          echo "PYTHON_VERSIONS=[$(IFS=,; echo "${versions[*]}")]" >> $GITHUB_ENV
```

Comment:
Let's break this down for clarify:
- `IFS=,` sets the Internal Field Separator to a comma for the subsequent command
- `echo "${versions[*]}"` prints all items in the array, separated by the IFS
- `echo "PYTHON_VERSIONS=[$(...)]"` puts the comma-separated string inside square brackets to form a valid JSON array
- `>> $GITHUB_ENV` appends the result to the GitHub environment, making the `PYTHON_VERSIONS` variable available to subsequent steps